estate/models/estate_property.py

```diff
@@ -0,0 +1,97 @@
+    description = fields.Text("Property description")
+    postcode = fields.Char("Postcode")
+    date_availability = fields.Date("Date of availability", default=lambda _self: fields.Date.add(fields.Date.today(), days=30), copy=False)
+    expected_price = fields.Float("Expected price", required=True)
+    selling_price = fields.Float("Selling price", readonly=True, copy=False)
+    bedrooms = fields.Integer("No. of bedrooms", default=2)
+    living_area = fields.Integer("Living area (sqm)")
+    facades = fields.Integer("No. of facades")
+    garage = fields.Boolean("Has a garage")
+    garden = fields.Boolean("Has a garden", default=False)
+    garden_area = fields.Integer("Garden area (sqm)", default=0)
+    garden_orientation = fields.Selection(
+        string="Garden orientation",
+        selection=[
+            ('north', "North"),
+            ('south', "South"),
+            ('east', "East"),
+            ('west', "West")
+            ],
+        default=None,
+        )
+    # Reserved
+    active = fields.Boolean("Active", default=True)
+    state = fields.Selection(
+        string="Status",
+        selection=[
+            ('new', "New"),
+            ('recieved', "Offer Received"),
+            ('accepted', "Offer Accepted"),
+            ('sold', "Sold"),
+            ('cancelled', "Cancelled")
+            ],
+        required=True,
+        copy=False,
+        default='new',
+        )
+    # Relational
+    property_type_id = fields.Many2one('estate.property.type', string="Type")
+    tag_ids = fields.Many2many('estate.property.tag', string="Tags")
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string="Offers")
+    buyer_id = fields.Many2one('res.partner', string="Buyer")
+    salesperson_id = fields.Many2one('res.users', string="Salesperson", default=lambda self: self.env.user)
+    # Computed
+    area_total = fields.Float("Total area", compute='_compute_area_total')
+    best_offer = fields.Float("Best offer", compute='_compute_best_offer')
+
+    def _compute_area_total(self):
+        for record in self:
+            record.area_total = record.living_area + record.garden_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        for record in self:
+            best = 0
+            for offer in record.offer_ids:
+                if offer.price > best:
+                    best = offer.price
+            record.best_offer = best
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = None
+
+    def action_set_sold(self):
+        for record in self:
+            if record.state != 'cancelled':
+                record.state = 'sold'
+            else:
+                raise UserError("The property is already cancelled.")
```

Comment:
```suggestion
                raise UserError(self.env._("The property is already cancelled."))
```
In Odoo, translatable strings in "data"-type content are automatically exported for translation. However, with other user-facing strings, Odoo cannot automatically know whether they are translatable. To ensure these strings are exported for translation, you need to explicitly mark them. This is achieved by wrapping the literal string with the `self.env._()` function. The `_()` function signals to Odoo's translation system that the string should be included in the translation catalog.

More info here: https://www.odoo.com/documentation/18.0/developer/howtos/translations.html#explicit-exports